SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -50,90 +55,157 @@ Plot {
+	// Currently only supports tick labels that don't extend past
+	// the side opposite their respective axis.
+	// TODO: this method could be refactored to use the showTickLabelsX/Y
+	// instance vars to skip corresponding calculations below when false.
+	prCalcLabelSpace { |viewRect|
+		var gridRect;
+		var xTkWd, yTkHt, tkLHang, tkBHang;
+		var xTkLHang, xTkTHang, xTkRHang, xTkBHang;
+		var yTkLHang, yTkTHang, yTkRHang, yTkBHang;
+		var tkBMargin, tkTMargin, bottomMargin, totalBottomPad, totalTopPad, totalVertSpace;
+		var tkLMargin, leftMargin, xLeftPad, xRightPad, leftPad, rightPad, totalHorizPad;
+
+		var htAllowsXLabels = true, htAllowsYLabels = true;
+		var wdAllowsXLabels = true, wdAllowsYLabels = true;
+		var sizeAllowsXLabels = true, sizeAllowsYLabels = true;
+		var xAxHt = 0, xAxMargin = 0, yAxWd = 0, yAxMargin = 0;
+
+		gridRect = viewRect.insetBy(borderMargin); // default with no labels
+
+		// Quick return if labels on both axes are disabled.
+		if (showTickLabelsX.not and: { showTickLabelsY.not }) {
+			^[gridRect, false, false]
 		};
 
-		// y axis label spacing
-		if(viewRect.width >= maxWidthWithLabel and: { ytkLabels.notNil }) {
-			lmargin = txtPad + ytkLabelSize.width + txtPad;
-			if (yLabelSize.height > 0) {
-				lmargin = lmargin + yLabelSize.height + txtPad };
-			if (hinset > 0) { // x labels present
-				rmargin = hinset;
-				lmargin = max(rmargin, lmargin);
-				hinset = (rmargin + lmargin) / 2;
-				hshift = max(0, lmargin - hinset);
-			} { // only y labels present
-				hinset = (lmargin + minGridMargin) / 2;
-				hshift = lmargin - hinset;
-			};
-			yLabelGridOffset = Point(lmargin, ytkLabelSize.height);
+		xTkWd = drawGrid.x.labelSize.width; // _tick_ label size
+		yTkHt = drawGrid.y.labelSize.height;
+
+		// Left, Top, Right, Bottom overhang space
+		#xTkLHang, xTkTHang, xTkRHang, xTkBHang = drawGrid.x.labelOverhang.max(0);
+		#yTkLHang, yTkTHang, yTkRHang, yTkBHang = drawGrid.y.labelOverhang.max(0);
+
+		if(labelX.notNil) {
+			xAxHt     = labelX.bounds(labelFont).height;
+			xAxMargin = labelMargin;
 		};
 
-		hinset = max(hinset, minGridMargin);
-		vinset = max(vinset, minGridMargin);
-		drawGrid.x.labelOffset = xLabelGridOffset;
-		drawGrid.y.labelOffset = yLabelGridOffset;
-		gridRect = viewRect.insetBy(hinset, vinset) + [hshift, vshift, 0, 0];
-		drawGrid.bounds = gridRect;
+		// First calculate vertical space needed for tick and axis labels
+		// to determine if X labels will be hidden based on the view height.
+		// (If so, Y labels that overhang vertically may need to also be hidden).
+
+		tkBHang = max(xTkBHang, yTkBHang);
+
+		// if there's label overhang, give it its margin
+		tkBMargin = if(tkBHang  > 0) { labelMargin } { 0 };
+		tkTMargin = if(yTkTHang > 0) { labelMargin } { 0 };
+
+		// only the largest margin of all elements is needed
+		bottomMargin = maxItem([xAxMargin, borderMargin, tkBMargin]);
+
+		// add up all elements extending below the grid
```

Comment:
Note the nonconventional alignment of `=` in this method.
It breaks from the style guidelines, but I find it easier to follow when there are so many variables with subtly different names and varying length.
If this is a problem I can remove the extra white space.

Reply:
My preference is to not include any manual non-standard formatting. I acknowledge that it looks better, but I believe even more strongly that (almost?) everything should be formatted automatically.
OTOH, the [automatic liner](https://github.com/supercollider/supercollider/actions/runs/3832847520/jobs/6523627143#step:3:59) for the class library files (which is set to not cause CI to fail since some existing .sc files don't conform) does not seem to catch this as an issue. So... maybe this is acceptable?

Reply:
This is interesting... I didn't know we were linting sc files. Ah ok i see it's just [checking](https://github.com/supercollider/supercollider/blob/develop/.editorconfig) tab character choice and length. In this case all of these `=`s align somewhere mid-line so the head of the line still conforms.

I'll defer to whoever feels strongest about this ;)

Reply:
OK I've done the style cleanup you requested 👍 https://github.com/supercollider/supercollider/pull/5942/commits/58e7e9594307a87381fe7592022ee71902ae4c67